config/reader.go

```diff
@@ -610,7 +610,24 @@ func processConfig(unprocessedConfig *Config, fromCloud bool, logger logging.Log
 		}
 	}
 
-	if err := logging.UpdateLoggerRegistry(cfg.LogConfig); err != nil {
+	appendedLogCfg := make([]logging.LoggerPatternConfig, 0, len(cfg.LogConfig)+len(cfg.Services)+len(cfg.Components))
+	appendedLogCfg = append(appendedLogCfg, cfg.LogConfig...)
+	for _, serv := range cfg.Services {
+		resLogCfg := logging.LoggerPatternConfig{
+			Pattern: "rdk." + serv.ResourceName().String(),
+			Level:   serv.LogConfiguration.Level.String(),
+		}
+		appendedLogCfg = append(appendedLogCfg, resLogCfg)
+	}
+	for _, comp := range cfg.Components {
+		resLogCfg := logging.LoggerPatternConfig{
+			Pattern: "rdk." + comp.ResourceName().String(),
+			Level:   comp.LogConfiguration.Level.String(),
+		}
+		appendedLogCfg = append(appendedLogCfg, resLogCfg)
+	}
+
+	if err := logging.RegisterConfig(appendedLogCfg); err != nil {
 		return nil, err
 	}
```

Comment:
nice!

the logic looks correct - two minor suggestions:

* add small doc-string describing the order of registration
* encapsulate this in a function (e.g. `registerLoggingConfig(unprocessedConfig *Config, ...)`), just because `processConfig` is pretty massive as is...

Reply:
done